python/deltalake/writer.py

```diff
@@ -202,6 +206,22 @@ def visitor(written_file: Any) -> None:
         )
 
 
+def __throw_error_if_delta_append_only_and_mode_not_append(
```

Comment:
nit: I think this function name could be a bit shorter. What do you think of `__enforce_append_only()`? I think the existing name would work well as a doc string.